Essentials/src/com/earth2me/essentials/commands/Commandeco.java

```diff
@@ -53,6 +54,9 @@ public void run(final Server server, final CommandSource sender, final String co
 
     @Override
     protected void updatePlayer(final Server server, final CommandSource sender, final User player, final String[] args) throws NotEnoughArgumentsException, ChargeException, MaxMoneyException {
+        if(isPercent && cmd != EcoCommands.RESET) {
+            amount = player.getMoney().multiply(amount.divide(new BigDecimal("100.0")));
```

Comment:
This can be optimized quite a bit.
```suggestion
            amount = player.getMoney().multiply(amount).scaleByPowerOfTen(-2);
```

This abuses how BigDecimals & Integers work to get around a division.